### PR TITLE
Fix missing control to force a filter update on pose updates

### PR DESCRIPTION
**Pull request by glpuga**
### Proposed changes

Major changes:
- Fixes lack of pose/transform update when setting a new initial pose.
- Add `request_nomotion_update` service (address #184)

Minor changes:
- Fix that ccache was not being used when building ROS 1.
- Rename `initialize_with_pose()` in `amcl_nodelet.xpp` to `reinitialize_with_pose()`, which is the name of the same function in `amcl_node.xpp`. The old name also overloaded the name of a different function.

#### Type of change

- [x] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

_Put an `x` in the boxes that apply. This is simply a reminder of what we will require before merging your code._

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga_amcl/test/filter_update_control/test_filter_update_control_mixin.cpp [hunk 1]
```diff
@@ -205,10 +234,29 @@ TEST_F(FilterUpdateControlTests, ResamplingPollerWithNPoliciesUsesShortCircuitEv
...
+    // iteration 9
+    EXPECT_CALL(uut, update_motion(_)).Times(1);
+    EXPECT_CALL(uut, sample()).Times(1);
+    EXPECT_CALL(uut, update_sensor(_)).Times(1);
+    EXPECT_CALL(uut, reweight()).Times(1);
+    EXPECT_CALL(uut, resample()).Times(1);
   }
 
+  auto next_force_update_flag_value = force_update_values.cbegin();
   for (const auto& iteration_result : expected_return_values) {
```
Comment (hidmic) [inline beluga_amcl/test/filter_update_control/test_filter_update_control_mixin.cpp:256]: @glpuga nit^2: you may want to use `ranges::views::zip` here (can't find better usage example than [this](https://stackoverflow.com/questions/61579631/ranges-v3-zip-with-auto-and-structured-bindings)).

File: beluga_system_tests/test/test_system.cpp [hunk 2]
```diff
@@ -120,7 +120,7 @@ TEST_P(BelugaSystemTest, testEstimatedPath) {
       }));
   for (const auto [iteration, data_point] : ranges::views::enumerate(test_data.data_points)) {
     auto scan_copy = data_point.scan;
-    const auto updated_pose = pf.update_filter(data_point.odom, std::move(scan_copy));
+    const auto updated_pose = pf.update_filter(data_point.odom, std::move(scan_copy), false);
```
Comment (hidmic) [inline beluga_system_tests/test/test_system.cpp:123]: @glpuga nit^2: consider using a named constant for readability sake:
```suggestion
    constexpr bool kForceUpdate = true;
    const auto updated_pose = pf.update_filter(data_point.odom, std::move(scan_copy), !kForceUpdate);
```

**Review by hidmic — APPROVED** [review]
Code LGTM at face value, but `codecov` ain't happy.

**Comment by glpuga** [comment]
Thanks for the review, will take a look in the evening.

File: beluga_amcl/test/filter_update_control/test_filter_update_control_mixin.cpp (hunk 1 above)
Comment (glpuga) [reply beluga_amcl/test/filter_update_control/test_filter_update_control_mixin.cpp:256]: fixed.

**Comment by glpuga** [comment]
> Code LGTM at face value, but `codecov` ain't happy.

The dislike is mutual.

Screenshot from 2023-07-22 16-43-50

**Pull request merged**